tools/refguide_check.py

```diff
@@ -792,57 +792,43 @@ def _run_doctests(tests, full_name, verbose, doctest_warnings):
     runner = DTRunner(full_name, checker=Checker(), optionflags=flags,
                       verbose=verbose)
 
-    output = []
+    output = io.StringIO()
     success = True
-    def out(msg):
-        output.append(msg)
-
-    class MyStderr:
-        """
-        Redirect stderr to the current stdout
-        """
-        def write(self, msg):
-            if doctest_warnings:
-                sys.stdout.write(msg)
-            else:
-                out(msg)
-
-        # a flush method is required when a doctest uses multiprocessing
-        # multiprocessing/popen_fork.py flushes sys.stderr
-        def flush(self):
-            if doctest_warnings:
-                sys.stdout.flush()
+    tmp_stderr = sys.stdout if doctest_warnings else output
 
-    # Run tests, trying to restore global state afterward
-    with np.errstate(), np.printoptions(), redirect_stderr(MyStderr()):
+    @contextmanager
+    def TempCwd():
```

Comment:
Nit: would probably stick to `snake_case` here

Reply:
I thought it was going to be a class, fixed now.